src/Controller/LoginController.php

```diff
@@ -33,6 +34,8 @@ public function login() : ?Response
 
         if (!$this->request->is('post')) {
             // Display login form.
+            $this->set('redirect', Hash::get($this->request->getQueryParams(), 'redirect'));
```

Comment:
`$this->request->getQuery('redirect')` should do the job without using `Hash` explicitly — it's still used, but internally.

Reply:
PS: Maybe it's possible to use `_view.request.getQuery('redirect')` directly in the template.

Reply:
Right, but on the doc block of `getQuery` I see:
```    
 * Developers are encouraged to use getQueryParams() when possible as it is PSR-7 compliant, and this method
 * is not.
 *
 * ### PSR-7 Alternative
 *
 * $value = Hash::get($request->getQueryParams(), 'Post.id', null);
 *
```

Reply:
Uhm… right. However I can see no deprecation notices or whatever. I'd say that this warning is mostly addressed to users developing a middleware that's meant to be compatible with other PHP frameworks and libraries.

If it's possible to use `_view.request.getQuery('redirect')` directly in the template, I'd go with that. Otherwise, if left in the controller, it's fine like this.

Reply:
ok, done in twig